### DISCUSSION TITLE
UI locale/translation

### DISCUSSION THREAD

**Discussion by anarion80** [heyform/heyform#89] [General]
Hello.

How does UI translation work? 
I have translated all locale/language/translation settings I could find: https://github.com/heyform/heyform/pull/88 to Polish.
I have my browser language set to Polish:
![image](https://github.com/heyform/heyform/assets/2185791/53a4c1d7-c599-4734-82fd-f4c02df9f3ea)

I am able to change Form language to Polish:
![image](https://github.com/heyform/heyform/assets/2185791/9a8c2a1d-770c-40c0-9619-38f2d704dd2c)

But I can't force the app to switch to Polish in the UI or in the form rendering.

I can see that in browser local storage, under `UserStore` key I have `lang: "en"` still. (Edit: changing this to "pl" in mongo db directly doesn't change anything)

What am I missing. What else is needed for the app to use my provided translations?

**Comment by iMuFeng** [comment]
You can convert 'pl-PL' to 'pl' in the detecting browser language function:

https://github.com/heyform/heyform/blob/a9f764e6a475ff9075e1297737c1850715112c22/packages/webapp/src/pages/form/views/FormComponents/utils/browser-language.ts#L1

**Reply by anarion80** [reply]
Sorry, not sure what you mean. There is no change even with 'pl':
![image](https://github.com/heyform/heyform/assets/2185791/5a053ed1-0baa-4485-8126-ff8b973c04cc)
Still no Polish language anywhere :(

**Reply by iMuFeng** [reply]
![image](https://github.com/heyform/heyform/assets/1407009/d6523966-e181-4e68-9197-b24c82856e6a)
Polish translation is working, it just needs to alias `pl-PL` with `pl`.

**Reply by anarion80** [reply]
Thanks for merging the PR! I still don't understand how to see that Polish locale in action 😢 🤣  Perhaps my building of docker image is off...
I will wait for a new official docker release (with the Polish translation included) and check then.

**Reply by iMuFeng** [reply]
![image](https://github.com/heyform/heyform/assets/1407009/89e1decc-5298-499c-bcfa-fd93e7e8a65e)
I will add a language switcher to the left sidebar.
